File: tests/fixtures/selenium-chaos-suite-csharp/src/Pages/OrdersPage.cs
using OpenQA.Selenium;

namespace SeleniumChaosSuite.Pages;

public class OrdersPage : BasePage
{
    public OrdersPage(IWebDriver driver) : base(driver) { }

    public async Task OpenOrdersRoute()
    {
        var current = Driver.Url;
        await Open($"{current.TrimEnd('/')}/orders");
    }

    public Task<string> GetBodyText()
    {
        return Task.FromResult(Driver.FindElement(By.CssSelector("body")).Text);
    }

    public Task TryFilterByStatus(string status)
    {
        var selects = Driver.FindElements(By.CssSelector("select"));
        if (selects.Count > 0)
        {
            selects[0].SendKeys(status);
        }

        return Task.CompletedTask;
    }

    public Task<string> OpenFirstOrderIfExists()
    {
        var links = Driver.FindElements(By.CssSelector("a"));
        if (links.Count > 0)
        {
            links[0].Click();
            Wait.Until(d => d.FindElement(By.CssSelector("body")).Displayed);
        }

        return Task.FromResult(Driver.Url);
    }
}

File: tests/fixtures/selenium-chaos-suite/src/pages/OrdersPage.ts
import { By } from "selenium-webdriver";
import { BasePage } from "./BasePage";
import { AssertionHelper } from "../core/AssertionHelper";
import { WaitUtils } from "../core/WaitUtils";

// Non-existent page but still asserted loosely.
export class OrdersPage extends BasePage {
  async openOrdersRoute() {
    const current = await this.driver.getCurrentUrl();
    await this.open(`${current.replace(/\/$/, "")}/orders`);
  }

  async assertOrdersPageOrFallback() {
    const body = await this.driver.findElement(By.css("body")).getText();
    const pass = body.includes("Order") || body.includes("404") || body.includes("Products");
    AssertionHelper.shouldBeTruthy(pass, "Expected order or fallback content");
  }

  async tryFilterByStatus(status: string) {
    const selects = await this.driver.findElements(By.css("select"));
    if (selects.length) {
      await selects[0].sendKeys(status);
      await WaitUtils.sleep(200);
    }
    AssertionHelper.shouldBeTruthy(true, "Forced pass regardless of filter result");
  }

  async openFirstOrderIfExists() {
    const links = await this.driver.findElements(By.css("a"));
    if (links.length) {
      await links[0].click();
      await WaitUtils.sleep(300);
    }
    const url = await this.driver.getCurrentUrl();
    AssertionHelper.shouldBeTruthy(url.length > 0, "URL should never be empty");
  }
}
